Modules/getpath.py

```diff
@@ -783,6 +783,19 @@ def search_up(prefix, *landmarks, test=isfile):
     base_exec_prefix = config.get('base_exec_prefix') or EXEC_PREFIX or base_prefix
 
 
+# ******************************************************************************
+# MISC. RUNTIME WARNINGS
+# ******************************************************************************
+
+# When running Python from the build directory, if libpython is dynamically
+# linked, the wrong library might be loaded.
+if build_prefix and not build_prefix.startswith(dirname(abspath(library))):
```

Comment:
`library` can be `None`, which is causing the failing CI:

```suggestion
if build_prefix and library and not build_prefix.startswith(dirname(abspath(library))):
```

It's probably worth stretching this case across multiple lines, though.